src/main/scala/com/iadvize/sbt/avro/AvroPlugin.scala

```diff
@@ -164,6 +166,32 @@ object AvroPlugin extends AutoPlugin {
     }
   }
 
+  lazy val compatibilityTask = Def.task {
+    val logger = streams.value.log
+
+    val configuredSchemaRegistryEndpoint = schemaRegistryEndpoint.value
+    val schemaRegistryClient = new CachedSchemaRegistryClient(configuredSchemaRegistryEndpoint, 10000)
+
+    val schemasToTestForCompatibility = parseSchemas(logger, resourceManaged.value.toPath) ++
```

Comment:
using `resourceManaged` as well means that if users generate their schemas from classes (using [avro4s](https://github.com/sksamuel/avro4s) for example) they can still use this task. It would be good if this could be added to the `upload` task as well as currently it only supports reading schemas from `resourceDirectory`. I was going to do this as another commit here but thought you guys may want it added as a separate PR. What do you guys think?

Reply:
Hi @lacarvalho91 !
Thanks for this new contribution :)
Yep, Like your said, I think this is maybe better to added it as a separate PR than an another commit. But, if it's simpler for you to add it as an another commit, please feel free to do it, no worries :)

Reply:
@jocelyndrean agreed. I'll do in another PR - will give me the chance to test out how well combining `avro4s` (for schema generation) and this plugin works and document it (if you guys are happy for me to do so) :)

Reply:
"if you guys are happy for me to do so" -> We are :D

Reply:
@jocelyndrean sweet! I'll raise another PR at some point soon with the change to the `upload` task then and some docs. This PR is ready then.